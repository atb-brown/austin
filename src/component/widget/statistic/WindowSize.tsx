import useWindowDimensions from "../../../hook/dimension/useWindowDimensions";
import React from "react";

/**
 * This component displays the width and height of the window.
 *
 * @return {ReactElement} React Component
 */
export default function WindowSize() {
  const { width, height } = useWindowDimensions();
  const title = "Window Size";
  const fontSize = Math.max(10, width / 130);
  return (
    <div
      style={{ display: "flex", flexDirection: "column", fontSize: fontSize }}
    >
      <div>{title}</div>
      <div>{width + " x " + height}</div>
    </div>
  );
}
