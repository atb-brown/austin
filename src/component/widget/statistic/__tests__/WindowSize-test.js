import WindowSize from "../WindowSize";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

it("renders to match snapshot", () => {
  const { asFragment } = render(<WindowSize />);

  expect(asFragment()).toMatchSnapshot();
});
