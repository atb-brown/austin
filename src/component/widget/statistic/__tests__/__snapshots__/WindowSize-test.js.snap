// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    style="font-size: 10px; display: flex; flex-direction: column;"
  >
    <div>
      Window Size
    </div>
    <div>
      1024 x 768
    </div>
  </div>
</DocumentFragment>
`;
